src/analytics/selectContentTracker/selectContentTracker.test.ts

```diff
@@ -0,0 +1,197 @@
+  const spy = jest.spyOn(SelectContentTracker.prototype, "pushToDataLayer");
+  const spyTrackNavigation = jest.spyOn(
+    SelectContentTracker.prototype,
+    "trackSelectContent",
+  );
+  const constructorSpy = jest.spyOn(
+    SelectContentTracker.prototype,
+    "initialiseEventListener",
+  );
+
+  test("new instance should call event listener", () => {
+    const instance = new SelectContentTracker();
+    expect(newInstance.initialiseEventListener).toBeCalled();
+  });
+
+  test("initialiseEventListener should attach event listener to each details element", () => {
+    const details1 = document.createElement("details");
+    const details2 = document.createElement("details");
+
+    document.body.appendChild(details1);
+    document.body.appendChild(details2);
+
+    // Spy on addEventListener for both elements
+    const addEventListenerSpy = jest.spyOn(
+      HTMLElement.prototype,
+      "addEventListener",
+    );
+
+    // Initialise event listeners
+    newInstance.initialiseEventListener();
+
+    expect(addEventListenerSpy).toHaveBeenCalledTimes(2); // Ensure it's called twice
+  });
+
+  test("toggle should call trackSelectContent", () => {
+    const details = document.createElement("details");
+    details.addEventListener("toggle", (event) => {
+      expect(newInstance.trackSelectContent).toBeCalled();
+    });
+  });
+  test("pushToDataLayer is called", () => {
+    const details = document.createElement("details");
+    details.addEventListener("toggle", (event) => {
+      newInstance.trackSelectContent(event);
+    });
+    details.dispatchEvent(action);
+    expect(newInstance.pushToDataLayer).toBeCalled();
+  });
+  test("should push data to data layer for each details component", () => {
+    const span1 = document.createElement("span");
+    span1.className = "govuk-details__summary-text";
+    span1.textContent = "Details Summary Text";
+    const firstDetails = document.createElement("details");
+    firstDetails.open = true;
+    firstDetails.append(span1);
+    const secondDetails = document.createElement("details");
+    secondDetails.open = true;
+    const span2 = document.createElement("span");
+    span2.className = "govuk-details__summary-text";
+    span2.textContent = "Details Summary Text 2";
+    secondDetails.append(span2);
+    const dataLayerEventFirstDetails: SelectContentEventInterface = {
+      event: "event_data",
+      event_data: {
+        event_name: "select_content",
+        type: "details ui",
+        url: "undefined",
+        text: "Details Summary Text",
+        section: "undefined",
+        action: "opened",
+        external: "undefined",
+        link_domain: "undefined",
+        "link_path_parts.1": "undefined",
+        "link_path_parts.2": "undefined",
+        "link_path_parts.3": "undefined",
+        "link_path_parts.4": "undefined",
+        "link_path_parts.5": "undefined",
+      },
+    };
+
+    const dataLayerEventSecondDetails: SelectContentEventInterface = {
```

Comment:
I think the only difference between `dataLayerEventFirstDetails` and `dataLayerEventSecondDetails` is the `text` attribute? So i think the first example is enough of a test, as there's not functionality to test the difference between

Reply:
Hmm but I was just checking that if there are multiple details components , it is retrieving the right values for each ?

Reply:
So essentially verifying this loop 
<img width="637" alt="Screenshot 2024-05-24 at 09 30 05" src="https://github.com/govuk-one-login/di-fec-ga4/assets/148252375/d4d2aed7-0fb0-45a8-a6ed-fd77b1e5a02b">